src/devices/cpu/tlcs900/tmp94c241.cpp

```diff
@@ -0,0 +1,2308 @@
+#define SFR_MSAR3 m_mem_start_reg[3]
+#define SFR_MSAR4 m_mem_start_reg[4]
+#define SFR_MSAR5 m_mem_start_reg[5]
+#define SFR_MAMR0 m_mem_start_mask[0]
+#define SFR_MAMR1 m_mem_start_mask[1]
+#define SFR_MAMR2 m_mem_start_mask[2]
+#define SFR_MAMR3 m_mem_start_mask[3]
+#define SFR_MAMR4 m_mem_start_mask[4]
+#define SFR_MAMR5 m_mem_start_mask[5]
+
+#define TIMER_0_IS_RUNNING	BIT(SFR_T8RUN, 0)
+#define TIMER_1_IS_RUNNING	BIT(SFR_T8RUN, 1)
+#define TIMER_2_IS_RUNNING	BIT(SFR_T8RUN, 2)
+#define TIMER_3_IS_RUNNING	BIT(SFR_T8RUN, 3)
+#define TIMER_4_IS_RUNNING	BIT(SFR_T16RUN, 0)
+#define TIMER_6_IS_RUNNING	BIT(SFR_T16RUN, 1)
+#define TIMER_8_IS_RUNNING	BIT(SFR_T16RUN, 2)
+#define TIMER_A_IS_RUNNING	BIT(SFR_T16RUN, 3)
+#define PRESCALER_IS_ACTIVE	BIT(SFR_T16RUN, 7)
+
+#define T0_INPUT_CLOCK			((SFR_T01MOD >> 0) & 0x03)
+#define T1_INPUT_CLOCK			((SFR_T01MOD >> 2) & 0x03)
+#define PWM1_INTERVAL_SELECTION	((SFR_T01MOD >> 4) & 0x03)
+#define TO1_OPERATING_MODE		((SFR_T01MOD >> 6) & 0x03)
+#define T2_INPUT_CLOCK			((SFR_T23MOD >> 0) & 0x03)
+#define T3_INPUT_CLOCK			((SFR_T23MOD >> 2) & 0x03)
+#define PWM2_INTERVAL_SELECTION	((SFR_T23MOD >> 4) & 0x03)
+#define T23_OPERATING_MODE		((SFR_T23MOD >> 6) & 0x03)
+#define T4_INPUT_CLOCK			((SFR_T4MOD >> 0) & 0x03)
+#define UC4_CLEAR				((SFR_T4MOD >> 2) & 0x01)
+#define T4_CAPTURE_TIMING		((SFR_T4MOD >> 3) & 0x03)
+#define CAP4IN					((SFR_T4MOD >> 5) & 0x01)
+#define T6_INPUT_CLOCK			((SFR_T6MOD >> 0) & 0x03)
+#define T8_INPUT_CLOCK			((SFR_T8MOD >> 0) & 0x03)
+#define TA_INPUT_CLOCK			((SFR_TAMOD >> 0) & 0x03)
+#define EQ4T4 BIT(SFR_T4FFCR, 2)
+#define EQ5T4 BIT(SFR_T4FFCR, 3)
+#define EQ6T6 BIT(SFR_T6FFCR, 2)
+#define EQ7T6 BIT(SFR_T6FFCR, 3)
+#define EQ8T8 BIT(SFR_T8FFCR, 2)
+#define EQ9T8 BIT(SFR_T8FFCR, 3)
+#define EQATA BIT(SFR_TAFFCR, 2)
+#define EQBTA BIT(SFR_TAFFCR, 3)
+
+// Field values for timer mode selection on TnMOD SFRs:
+#define MODE_8BIT_TIMER		0
+#define MODE_16BIT_TIMER	1
+#define MODE_PPG			2
+#define MODE_PWM			3
+
+#define UPCOUNTER_0		m_timer[0]
+#define UPCOUNTER_1		m_timer[1]
+#define UPCOUNTER_2		m_timer[2]
+#define UPCOUNTER_3		m_timer[3]
+#define UPCOUNTER_4		m_timer16[0]
+#define UPCOUNTER_6		m_timer16[1]
+#define UPCOUNTER_8		m_timer16[2]
+#define UPCOUNTER_A		m_timer16[3]
+#define SFR_TREG0	m_t8_reg[0]
+#define SFR_TREG1	m_t8_reg[1]
+#define SFR_TREG2	m_t8_reg[2]
+#define SFR_TREG3	m_t8_reg[3]
+#define SFR_TREG4	m_t16_reg[0]
+#define SFR_TREG5	m_t16_reg[1]
+#define SFR_TREG6	m_t16_reg[2]
+#define SFR_TREG7	m_t16_reg[3]
+#define SFR_TREG8	m_t16_reg[4]
+#define SFR_TREG9	m_t16_reg[5]
+#define SFR_TREGA	m_t16_reg[6]
+#define SFR_TREGB	m_t16_reg[7]
+#define TIMER_CHANGE_0	m_timer_change[0]
+#define TIMER_CHANGE_1	m_timer_change[1]
+#define TIMER_CHANGE_2	m_timer_change[2]
+#define TIMER_CHANGE_3	m_timer_change[3]
+#define TIMER_CHANGE_4	m_timer_change[4]
+#define TIMER_CHANGE_6	m_timer_change[5]
+#define TIMER_CHANGE_8	m_timer_change[6]
+#define TIMER_CHANGE_A	m_timer_change[7]
```

Comment:
But they do add value over the expansions!

It is much easier to understand that some code is dealing with "Special Function Register TREG B" (hence the name `SFR_TREGB`) rather than, "some register whose value we keep in the position 7 of an array of 16-bit timer registers" (`m_t16_reg[7]`)